lessons/06-tool-calling/README.md

```diff
@@ -0,0 +1,426 @@
+}
+
+function getBackgroundOnCharacter(character) {
+    // Perform the task of getting background information on a character
+    // Return the background information
+    return `Background information on ${character}`;
+}
+
+const getBackgroundOnCharacterJson = {
+  name: "get-background-on-character",
+  description: "Get background information on a character",
+  parameters: {
+    type: "object",
+    properties: {
+      name: {
+        type: "string",
+        description: "The name of the character",
+      }
+    },
+    required: ["lat", "long"],
+  },
+  output: { type: "string" }
+};
+
+const findLandingSpotJson = {
+  name: "find-landing-spot",
+  description: "Finds a suitable landing spot",
+  parameters: {
+    type: "object",
+    properties: {
+      lat: {
+        type: "number",
+        description: "The latitude of the location",
+      },
+      long: {
+        type: "number",
+        description: "The longitude of the location",
+      },
+    },
+    required: ["lat", "long"],
+  },
+  output: { type: "object", properties: { lat: "number", long: "number" } }
+};
+
+
+const messages = [{ 
+    role: "user", 
+    content: `Tell me about Amelia Earhart`,
+}];
+
+const completion = await openai.chat.completions.create({
+    model: 'gpt-4o-mini',
+    messages: messages,
+    functions: [getBackgroundOnCharacterJson, findLandingSpotJson]
+  });
+```
+
+**Time beetle**: "In the preceding code snippet we:" 
+
+- Define the metadata for the `find-landing-spot` tool and the `get-background-on-character` tool. 
+- Provide this metadata to the `client.getChatCompletions` call as part of the `functions` parameter. This tells the AI model that these tools are available to be called."
+
+**You**: "Got it, so the AI model will call the appropriate if I provide a prompt that matches to the tool's description?"
+
+**Time beetle**: "Almost, it will tell you what tool it thinks you should call and it will provide you with the parsed input parameters, but you need to call the tool yourself, let me show you how."
+
+### Calling a tool
+
+**Time beetle**: "As I was saying, the AI model will tell you what tool it thinks you should call and it will provide you with the parsed input parameters. You then need to call the tool yourself. Here's what the workflow will look like step by step:
+
+1. Wire up the tool call
+
+   First, you need to wire up the tool call in your code. This involves creating the function and a metadata representation of the tool, and then providing the metadata to the AI model.
+
+1. User makes a request via a prompt:
+   - Program makes a chat completion request to the AI model with the user prompt and tools metadata provided.
+   - Program receives a response from the AI model with the tool call and parsed input parameters if it thinks a tool should be called.
+   - If so, the developer interprets the response and calls the tool based on the function call provided by the AI model.
```

Comment:
Slightly different phrasing to consider:

If so, the developer interprets the response and invokes the tool based on the function call provided by the AI model.